src/Hyperion.Tests/UnsafeDeserializationExclusionTests.cs

```diff
@@ -22,5 +24,52 @@ public void CantDeserializeANaughtyType()
                     serializer.Deserialize<DirectoryInfo>(stream));
             }
         }
+
+        internal class ClassA
+        { }
+        
+        internal class ClassB
+        { }
+        
+        internal class ClassC
+        { }
+        
+        [Fact]
+        public void TypeFilterShouldThrowOnNaughtyType()
```

Comment:
We would have to eventually, but this can't be tested against Akka.NET from Hyperion side until we do a release of Hyperion since Akka.NET is dependent on Hyperion, not the other way around.
It would make more sense to test this on the Akka.NET side instead of on the Hyperion side.

Reply:
Based on my discussion with @Aaronontheweb, we could create an integration test on the Hyperion test suite so that changes are thoroughly tested against Akka.NET before Nuget package is released. The benefit is that when we fail to capture it in code review, the test will at least assist with that.

Reply:
> to eventually, but this can't be tested against Akka.NET from Hyperion side until we do a release of Hyperion since Akka.NET is dependent on Hyperion, not the other way around.
> It would make more sense to test this on the Akka.NET side instead of on the Hyperion

What @eaba is asking is that this test doesn't cover the same `Deserialze<object>` routine we use in Akka.NET itself and that could be added to this test suite here. We will have to do a separate PR for Akka.NET's consumption of this feature, of course, but we should also test that the way Akka.NET uses Hyperion is supported by this feature prior to releasing this package. Does that make sense @Arkatufus ?

Reply:
@Arkatufus handled this further down the PR